dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -3467,11 +3470,23 @@ RtpsUdpDataLink::RtpsWriter::gather_nack_replies_i(MetaSubmessageVec& meta_subme
                    "resend data %q-%q\n", pos->first.getValue(),
                    pos->second.getValue()));
       }
-      proxy.resend_i(*pos);
+      for (SequenceNumber seq = pos->first; seq <= pos->second; ++seq) {
+        const AddrSet& uni = consolidated_recipients_unicast[seq];
+        const AddrSet& multi = consolidated_recipients_multicast[seq];
+        const RtpsUdpSendStrategy::OverrideToken ot =
+          link->send_strategy()->override_destinations(multi.size() * 2 < uni.size() ? multi : uni);
```

Comment:
```suggestion
          link->send_strategy()->override_destinations(multi.size() * 2 > readers_.size() ? multi : uni);
```

Reply:
What's the goal here, fewer sends, right? But with a slight preference for unicast since it's not as much of a network hog?

Reply:
Assume we had one more map `req_count` from requested sequence number to number of remotes that requested it.  
`f = req_count[sn] / readers_.size()` is the fraction of remotes that requested `sn`.
If `f > .5`, then I would use multicast to send the packet and use unicast otherwise.